lesson_16/objects/objects_app/src/main/java/com/codedifferently/lesson16/Person/Person.java

```diff
@@ -0,0 +1,133 @@
+package com.codedifferently.lesson16.person;
+
+import java.util.ArrayList;
+
+public class Person {
+
+  private String name;
+  private String sex;
+  private String race;
+  private int age;
+  private boolean alive;
+  private double height;
+  private ArrayList<String> hairColor;
+
+  public Person(
+      String name,
+      String sex,
+      String race,
+      int age,
+      double height,
+      ArrayList<String> hairColor,
+      boolean alive) {
+    this.name = name;
+    this.sex = sex;
+    this.race = race;
+    this.age = age;
+    this.height = height;
+    this.hairColor = hairColor;
+    this.alive = alive;
+  }
+
+  public String getName() {
+    return name;
+  }
+
+  public void setName(String name) {
+    this.name = name;
+  }
+
+  public String getSex() {
+    return sex;
+  }
+
+  public void setSex(String sex) {
+    this.sex = sex;
+  }
+
+  public String getRace() {
+    return race;
+  }
+
+  public void setRace(String race) {
+    this.race = race;
+  }
+
+  public int getAge() {
+    return age;
+  }
+
+  public void setAge(int age) {
+    this.age = age;
+  }
+
+  public double getHeight() {
+    return height;
+  }
+
+  public void setHeight(double height) {
+    this.height = height;
+  }
+
+  public ArrayList<String> getHairColor() throws PersonIsBaldException {
+    if (hairColor == null || hairColor.isEmpty()) {
+      this.hairColor = new ArrayList<>();
+      hairColor.add("Bald");
+      throw new PersonIsBaldException("No hair color? This person must be bald!");
+    }
+    return hairColor;
+  }
+
+  public void setHairColor(ArrayList<String> hairColor) {
+    this.hairColor = hairColor;
+  }
+
+  public ArrayList<String> getNaturalHairColor() {
```

Comment:
This method works. But for future reference, this currently returns only one match due to the `break` statement, even though the return type is a `list`, implying multiple matches are expected. 

Also, the list of hair colors is hardcoded. A more flexible approach such as allowing other colors would accommodate to real-life context, maybe by passing the list as a parameter or using a more dynamic approach.